web-common/src/main/java/edu/unc/lib/boxc/web/common/services/LorisContentService.java

```diff
@@ -80,42 +81,43 @@ public void getMetadata(String simplepid, String datastream, OutputStream outStr
         path.append(idToPath(simplepid, 4, 2))
                 .append(simplepid).append(".jp2").append("/info.json");
 
-        HttpGet method = new HttpGet(path.toString());
-        try (CloseableHttpResponse httpResp = httpClient.execute(method)) {
-            int statusCode = httpResp.getStatusLine().getStatusCode();
-            if (statusCode == HttpStatus.SC_OK) {
-                if (response != null) {
-                    response.setHeader("Content-Type", "application/json");
-                    response.setHeader("content-disposition", "inline");
-
-                    ObjectMapper iiifMapper = new IiifObjectMapper();
-
-                    ImageService respData = iiifMapper.readValue(httpResp.getEntity().getContent(),
-                            ImageService.class);
-                    respData.setIdentifier(new URI(URIUtil.join(basePath, "jp2Proxy", simplepid, "jp2")));
-
-                    HttpEntity updatedRespData = EntityBuilder.create()
-                            .setText(iiifMapper.writeValueAsString(respData))
-                            .setContentType(ContentType.APPLICATION_JSON).build();
-                    httpResp.setEntity(updatedRespData);
-
-                    FileIOUtil.stream(outStream, httpResp);
-                }
-            } else {
-                if ((statusCode == 500 || statusCode == 404) && retryServerError > 0) {
-                    getMetadata(simplepid, datastream, outStream, response, retryServerError - 1);
-                } else {
-                    LOG.error("Unexpected failure: {}", httpResp.getStatusLine());
-                    LOG.error("Path was: {}", method.getURI());
+        int statusCode = -1;
+        String statusLine = null;
+        do {
```

Comment:
i'm not familiar with do loops, how does it know when to stop the loop?

Reply:
its the same as a while loop, except that it checks the condition at the end of the iteration rather than the beginning. See line 119.